src/iocore/hostdb/HostDBInfo.cc

```diff
@@ -73,7 +73,7 @@ HostDBInfo::assign(SRV const *srv, char const *name) -> self_type &
   data.srv.srv_priority = srv->priority;
   data.srv.srv_port     = srv->port;
   data.srv.key          = srv->key;
-  data.srv.srv_offset   = reinterpret_cast<char const *>(this) - name;
+  data.srv.srv_offset   = name - reinterpret_cast<char const *>(this);
```

Comment:
Now that you figured this out, can you please add a comment explaining this pointer arithmetic? Or is there a comment elsewhere? The calling logic which loops over `rr_info` is not trivial for me to think through.

Reply:
I'll add a comment.  The bigger issue here is that the code assumes that the pointer parameter passed in is one that can be recreated from a 16-bit offset value.  You couldn't, for example, pass in a pointer into the stack.  This is a huge design flaw.  While this change corrects the math, it doesn't address the underlying issue.